docs/datasets/index.rst

```diff
@@ -1,5 +1,6 @@
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
```

Comment:
Why change to 3 here? If we would like to merge detection and segmentation doc, we can further modify it in new PRs.